hack/cleanup.sh

```diff
@@ -1,7 +1,117 @@
-#!/bin/sh
+#!/bin/bash
+#set -xe 
+WAIT_FOR_CONTAINER="1"
+SUDO=''
+if (( $EUID != 0 )); then
+    SUDO='sudo'
+fi
 
-pkill -9 conmon
-mount |grep overlay |awk '{print $3}' |xargs umount
-mount |grep kubelet |awk '{print $3}' |xargs umount
-pkill -9 pause
-rm /var/lib/etcd -rf
+V()
+{
+  echo "### $*"
+  $*
+}
+
+
+function check_for_running_containers(){
+  if [[ ! -z $(${SUDO} crictl ps  | grep -v CONTAINER) ]];
+  then 
+    CONTAINERS=$(${SUDO} crictl ps | awk '{if(NR>1) print $1}')
+    for CONTAINER in $CONTAINERS
+    do
+      V ${SUDO} crictl stop $CONTAINER
+      sleep ${WAIT_FOR_CONTAINER}s
+    done
+  fi 
+}
+
+function remove_all_stopped_containers(){
+  if [[ ! -z $(${SUDO} crictl ps -q -a | grep -v CONTAINER) ]];
+  then 
+    CONTAINERS=$(${SUDO} crictl ps -a | awk '{if(NR>1) print $1}')
+    for CONTAINER in $CONTAINERS
+    do
+      V ${SUDO} crictl rm $CONTAINER 
+    done
+  fi 
+}
+
+function check_for_running_pods(){
+  if [[ ! -z $(${SUDO} crictl pods | grep -v "POD ID") ]];
+  then 
+    PODS=$(${SUDO} crictl pods | awk '{if(NR>1) print $1}')
+    for POD in $PODS
+    do
+      V ${SUDO} crictl stopp $POD 
+    done
+  fi 
+}
+
+function remove_all_pods(){
+  if [[ ! -z $(${SUDO} crictl pods | grep -v "POD ID") ]];
+  then 
+    PODS=$(${SUDO} crictl pods | awk '{if(NR>1) print $1}')
+    for POD in $PODS
+    do
+      V ${SUDO} crictl rmp $POD 
+    done
+  fi 
+}
```

Comment:
I believe these functions could be consolidated into a single line, killing and removing all containers:

`$ until crictl rmp --force --all; do sleep 1; done`